include/basicstream.hpp

```diff
@@ -43,9 +44,9 @@ class BasicStream {
     }
 
     const T& current(int32_t offset = 0) const
-        { return (offset < 0 && m_current + offset < m_current) || (offset >= 0 && m_current + offset < m_data.size()) ? m_data.at(m_current + offset) : m_data.back(); }
+        { return std::cmp_less(m_current + offset, m_data.size()) && std::cmp_greater_equal(m_current + offset, 0) ? m_data.at(m_current + offset) : m_data.back(); }
     T& current(int32_t offset = 0)
-        { return (offset < 0 && m_current + offset < m_current) || (offset >= 0 && m_current + offset < m_data.size()) ? m_data.at(m_current + offset) : m_data.back(); }
+        { return std::cmp_less(m_current + offset, m_data.size()) && std::cmp_greater_equal(m_current + offset, 0) ? m_data.at(m_current + offset) : m_data.back(); }

```

Comment:
I didn't know these functions :). But it doesn't fix the problem, because `m_current + offset` (unsigned + signed) was the root cause and is still performed. e.g. `1u - 2 == 4294967295u` (underflow).

I don't see a better-looking solution than what I proposed. It would also be fine to leave as-is, because in practice it will not fail unless size is very large.

https://godbolt.org/z/z9bdb533f

Reply:
Right, I missed that. Casting `m_current + offset` to int will do the trick.